Sources/BaggageLogging/BaggageContextLogging.swift

```diff
@@ -0,0 +1,125 @@
+    /// ### Implementation note
+    ///
+    /// Implementors, such as libraries or frameworks, which conform to `BaggageContextLogging` with their "Framework Context"
+    /// types, should implement this logger by wrapping the "raw" logger associated with this context with the `logger.with(BaggageContext:)` function,
+    /// which efficiently handles the bridging of baggage to metadata context values.
+    var logger: Logger { get set }
+}
+
+public struct DefaultContext: BaggageContextLogging {
+    public var baggage: BaggageContext
+    public var logger: Logger
+
+    public init(context baggage: BaggageContext, logger: Logger) {
+        self.baggage = baggage
+        self.logger = logger.with(context: baggage)
+    }
+}
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: BaggageContext (as additional Logger.Metadata) LogHandler
+
+/// Proxying log handler which adds `BaggageContext` as metadata when log events are to be emitted.
+public struct BaggageMetadataLogHandler: LogHandler {
+    var underlying: Logger
+    var context: BaggageContext
+
+    public var logLevel: Logger.Level {
+        get {
+            self.underlying.logLevel
+        }
+        set {
+            self.underlying.logLevel = newValue
+        }
+    }
+
+    public func log(level: Logger.Level,
+                    message: Logger.Message,
+                    metadata: Logger.Metadata?,
+                    source: String,
+                    file: String,
+                    function: String,
+                    line: UInt) {
+        guard self.underlying.logLevel <= level else { return }
+
+        var effectiveMetadata = self.baggageAsMetadata()
+        if let metadata = metadata {
+            effectiveMetadata.merge(metadata, uniquingKeysWith: { _, r in r })
+        }
+        self.underlying.log(level: level, message, metadata: effectiveMetadata, source: source, file: file, function: function, line: line)
+    }
+
+    public var metadata: Logger.Metadata {
+        get {
+            [:] // FIXME: or good actually?
+        }
+        set {
+            newValue.forEach { k, v in
+                self.underlying[metadataKey: k] = v
+            }
+        }
+    }
+
+    private func baggageAsMetadata() -> Logger.Metadata {
+        var effectiveMetadata: Logger.Metadata = [:]
+        self.context.forEach { key, value in
+            if let convertible = value as? String {
+                effectiveMetadata[key.name] = .string(convertible)
+            } else if let convertible = value as? CustomStringConvertible {
+                effectiveMetadata[key.name] = .stringConvertible(convertible)
+            } else {
+                effectiveMetadata[key.name] = .stringConvertible(BaggageValueCustomStringConvertible(value))
+            }
+        }
+        return effectiveMetadata
+    }
+
+    struct BaggageValueCustomStringConvertible: CustomStringConvertible {
```

Comment:
Can this be `private` instead?

Reply:
I think internal is fine tbh, did not think through if making usableFromInline makes sense or not tho